### ISSUE TITLE
accessibility - keyboard locks up every minute or so using most recent insider build

### ISSUE THREAD

**Issue by RichCaloggero** [microsoft/vscode#169000]
Type: <b>Bug</b>

Windows 10, screen reader is NVDA, most recent insider build, as of December 13, 2022.

The first time I start vscode for the day, after a short time (about 3 minutes), my screen reader stops speaking. If I press nvdaKey+tab (read current focus), it says "setup", but nothing else and I can't do anything.

If I cancel out, I get back to vscode, but now every minute or so, my keyboard locks up, again with no speech. The only way now to get things going again is to switch away from vscode via alt+tab, then back again.

I cannot see the screen, so cannot give any more info, but I'll bet whatever that setup window that came up first was, it's coming up again...

Sorry I can't be more specific...

VS Code version: Code - Insiders 1.75.0-insider (11238faea62d570d77afe6edfe05c8b732c44a2b, 2022-12-13T05:22:12.417Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4800MQ CPU @ 2.70GHz (8 x 2694)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.91GB (11.26GB free)|
|Process Argv|--crash-reporter-id da8aa860-ae68-41ac-aa82-9caa04dc7fc1|
|Screen Reader|yes|
|VM|0%|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30422396
pythontb:30258533
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805:30301674
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30404738
cppdebug:30492333
vsclangdf:30492506
c4g48928:30535728
dsvsc012:30540252
pynewext54:30618038
pylantcb52:30590116
pyindex848:30611229
nodejswelcome1:30587009
pyind779:30611226
dbltrim-noruby:30604474
pythonsymbol12cf:30622697
fim-prod:30623723

```

</details>

**Comment by meganrogge** [comment]
I cannot reproduce this.

I am wondering if you are not in `Screen Reader Optimized` mode. Can you check the value of `editor.accessibilitySupport` in your settings and make sure it's set to `on`?

**Comment by steverep** [comment]
I can verify this issue, also with NVDA on Windows 10.  It occurs both with the latest insiders and with the production 1.74.0.  And yes accessibility support is enabled.

The symptom is usually that the screen reader simply stops reading anything after a period of time, and it can be temporarily fixed by minimizing the window or switching to another then going back to vscode.

I've also had the symptom where it will keep reading the same line after consecutive down or up arrow presses even though the cursor is clearly moving to the next or previous line.

**Comment by deepak1556** [comment]
Symptoms point to running with [AutoDisableAccessibility](https://github.com/microsoft/vscode/blob/342a1f4f52ff8521cd6860a95a4408e51694dfdc/src/main.js#L242-L245) experiment as the cause.

Can you try starting with `--disable-features=AutoDisableAccessibility` and see if the issue persists ?

**Comment by steverep** [comment]
Disabling that feature seems to have no effect.  The problem persists.

I'm also noticing that I can get NVDA to start reading again if I switch context (e.g. stops reading in file editor, switch to terminal then back, reading usually resumes).

**Comment by deepak1556** [comment]
Running new build https://monacotools.visualstudio.com/Monaco/_build/results?buildId=196804&view=results with potential fix for testing, will update here once the builds are available.

**Comment by deepak1556** [comment]
Can you test if the issue persists with the following insiders build, please make sure turn off updates temporarily in your settings with `"update.mode": "none"`

[Windows x64 Build Link](https://az764295.vo.msecnd.net/insider/e4bdaa1ea58f3f56cef36cd2d89d4ed1e5c9d443/VSCodeUserSetup-x64-1.75.0-insider.exe)

**Comment by Yeldece** [comment]
These are terrible bugs i've encountered on production 1.74.
The most annoying one for me was NVDA were trying to read whole document from top to bottom when ever I press up or down arrow.
With the given insider 1.75 link above, it looks fine for now. It also feels a bit faster than the usual vscode, out of topic.

**Comment by isidorn** [comment]
Thanks for filling this issue. I tried to reproduce with latest Insiders, NVDA on my Win10 VM and I could not repro, here are my steps. 
1. Turn on NVDA, VS Code insiders
2. Open the https://github.com/microsoft/vscode repo in VS Code Insiders
3. Open this file in VS Code Insiders https://github.com/microsoft/vscode-distro/blob/isidorn/charming-cuckoo/src/vs/workbench/contrib/debug/browser/statusbarColorProvider.ts#L1
4. Use up / down arrow for 7-8 minutes inside the file. Notice how NVDA reads everything even after all that time 

@RichCaloggero @steverep @brunoprietog can you tell me what am I missing in this steps? Also can you try with the build @deepak1556 provided [in this comment](https://github.com/microsoft/vscode/issues/169000#issuecomment-1350678203). That would really help us to understand this better since we cannot yet reproduce the problem on our side.

**Comment by brunoprietog** [comment]
I have not tested with insider, but in stable, if you press ctrl + shift + e to go to the file explorer, leave it still for a while, you can no longer do anything and it is necessary to change window. When I go back to the VS Code window, sometimes I can't read the lines, it reads the whole file from the beginning, and besides that, NVDA takes a long time to move through some controls. For example, when expanding a folder in the file explorer. All this is solved if I restart NVDA. But it is something I have to be doing very frequently to be able to work well.
As I was saying, this happens to me in the latest stable version. I will try the above solution. Thanks

**Comment by brunoprietog** [comment]
The above solution solves the performance problems I have been having for a long time when starting VS Code. However, when I leave the focus on the file browser for a minute or so, the problem persists and nothing else can be done. If I press ctrl + shift + e and switch contexts, the focus returns, but now I have lower performance with NVDA, which causes me to have to restart the screen reader anyway. Thanks

**Comment by brunoprietog** [comment]
After some time, I have been able to verify that it is still not resolved, but it does not happen in all cases.

The performance issues in VS Code at startup seems to be the only thing that was fixed, apparently. I hope you can keep and merge the changes from the above proposed solution because now I don't have to restart the screen reader after starting VS Code, which is much nicer.

However, the other two problems still persist.

Standing still for about a minute in the file explorer results in not being able to focus on anything.

The line problem I can now only reproduce with an extension, but it uses the same Monaco editor as VS Code. Install eamodio.gitlens extension. Now, focus the file explorer and look for a file with the arrows. Press shift f10 and scroll down to open file history. Then, choose any commit and press enter, then f6 to go to the editor that opened. There the lines do not read well. For example, given this file from a Ruby on Rails application:

```rb
# frozen_string_literal: true

# This file is used by Rack-based servers to start the application.

require_relative 'config/environment'

run Rails.application
Rails.application.load_server
```

Pressing up arrow or down arrow always reads this, except in the last line that reads blank:

```
# frozen_string_literal: true # This file is used by Rack-based servers to start the application. require_relative 'config/environment' run Rails.application Rails.application.load_server
```

Thanks

**Comment by RichCaloggero** [comment]
I just installed the build posted above:

- VSCodeUserSetup-x64-1.75.0-insider.exe

and now, everything works as expected. The screen reader no longer locks up on me every minute or so! This is great...

Hopefully, this will give the team enough information to track down this bug!

I will leave updateMode set to "none" until I hear that this bug has been squashed.

Thanx for the fix!

-- Rich

**Comment by brunoprietog** [comment]
I investigated further and the problem was indeed solved. While changing "update.mode" to none VS Code downloaded the update. I uninstalled and reinstalled and everything is fine. Thank you very much!

I hope you can release this solution quickly, as I can't use this specific code-insiders version with WSL. Try to install but I get 404 error.

**Comment by steverep** [comment]
@deepak1556 I agree that your test build seems to have solved the issue and navigation feels crisper and more responsive.  Please update insiders and also a patch release to 1.74 since this does not work with WSL or devContainers.

**Comment by deepak1556** [comment]
@RichCaloggero @steverep @brunoprietog thanks for confirming the fix, it will be available in Insiders today. Can you also respond to @isidorn comment in https://github.com/microsoft/vscode/issues/169000#issuecomment-1350796599 about any missing steps which will help us on our end to verify the fix. Thanks!


> also a patch release to 1.74 since this does not work with WSL or devContainers.

We just had a patch release yesterday `1.74.1`, you should be able to use the normal insiders with the fix on both WSL and devcontainers until we make a `1.74.2` release. Also, we will need the steps to verify if we want to make a patch release to stable. I will update here again once the fix is available with Insiders.

**Comment by deepak1556** [comment]
Insiders has been released with the fix, you can get insiders from https://code.visualstudio.com/insiders

**Comment by brunoprietog** [comment]
the steps are in this [comment](https://github.com/microsoft/vscode/issues/169000#issuecomment-1351681377), for the line reading and freezing problem

Thanks for the solution!

**Comment by RichCaloggero** [comment]
FOr me, the problem would occur when I was simply navigating the interface. It was most apparent, and seemed to be triggered more quickly, when navigating a menu like view (alt+v, then up/down arrows).

Question: I'm seeing behavior that I've never seen in other menus -- if I open view with alt+v, navigate via downArrow, escape out of the menu, then come back in with alt+v, it almost never focuses on the first menu item, but somewhere in the middle. Is this intentional?

Thanx for all the work on this!

-- Rich

**Comment by steverep** [comment]
> Can you also respond to @isidorn comment in [#169000 (comment)](https://github.com/microsoft/vscode/issues/169000#issuecomment-1350796599) about any missing steps which will help us on our end to verify the fix. Thanks!
 
I cannot say I noticed a repeatable pattern, but it was much more likely to occur when navigating around the UI in areas like the explorer, settings, extensions, menus, etc. than just staying within a single file.  So it doesn't surprise me that @isidorn's steps did not reproduce the issue.

**Comment by isidorn** [comment]
Closing since we have pushed a fix for this.
Adding a verified label based on users confirming the issue is indeed fixed. 

If some users can it would also be helpful to verify with the stable build that is about to come out in the next 24 hours. Here's the links to the build depending on your platform


darwin: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/VSCode-darwin.zip
darwin-arm64: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/VSCode-darwin-arm64.zip
darwin-universal: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/VSCode-darwin-universal.zip
linux-arm64: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/code-stable-arm64-1671532745.tar.gz
linux-armhf: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/code-stable-armhf-1671532441.tar.gz
linux-deb-arm64: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/code_1.74.2-1671532382_arm64.deb
linux-deb-armhf: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/code_1.74.2-1671532193_armhf.deb
linux-deb-x64: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/code_1.74.2-1671533413_amd64.deb
linux-rpm-arm64: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/code-1.74.2-1671532468.el7.aarch64.rpm
linux-rpm-armhf: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/code-1.74.2-1671532262.el7.armv7hl.rpm
linux-rpm-x64: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/code-1.74.2-1671533504.el7.x86_64.rpm
linux-snap-x64: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/code-stable-x64-1671534069.snap
linux-x64: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/code-stable-x64-1671533708.tar.gz
win32: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/VSCodeSetup-ia32-1.74.2.exe
win32-archive: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/VSCode-win32-ia32-1.74.2.zip
win32-arm64: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/VSCodeSetup-arm64-1.74.2.exe
win32-arm64-archive: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/VSCode-win32-arm64-1.74.2.zip
win32-arm64-user: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/VSCodeUserSetup-arm64-1.74.2.exe
win32-user: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/VSCodeUserSetup-ia32-1.74.2.exe
win32-x64: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/VSCodeSetup-x64-1.74.2.exe
win32-x64-archive: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/VSCode-win32-x64-1.74.2.zip
win32-x64-user: https://az764295.vo.msecnd.net/stable/e8a3071ea4344d9d48ef8a4df2c097372b0c5161/VSCodeUserSetup-x64-1.74.2.exe

**Comment by steverep** [comment]
Yes, it's also fixed in 1.74.2.
